vars/mtaBuild.groovy

```diff
@@ -75,106 +53,53 @@ void call(Map parameters = [:]) {
             .dependingOn('mtaBuildTool').mixin('dockerImage')
             .use()
 
-        new Utils().pushToSWA([
-            step: STEP_NAME,
-            stepParamKey1: 'scriptMissing',
-            stepParam1: parameters?.script == null
-        ], configuration)
-
-        dockerExecute(
-            script: script,
-            dockerImage: configuration.dockerImage,
-            dockerEnvVars: configuration.dockerEnvVars,
-            dockerOptions: configuration.dockerOptions,
-            dockerWorkspace: configuration.dockerWorkspace
-        ) {
-
-            String projectSettingsFile = configuration.projectSettingsFile?.trim()
-            if (projectSettingsFile) {
-                if (projectSettingsFile.startsWith("http")) {
-                    projectSettingsFile = downloadSettingsFromUrl(this, projectSettingsFile, 'project-settings.xml')
-                }
-                sh 'mkdir -p $HOME/.m2'
-                sh "cp ${projectSettingsFile} \$HOME/.m2/settings.xml"
-            }
-
-            String globalSettingsFile = configuration.globalSettingsFile?.trim()
-            if (globalSettingsFile) {
-                if (globalSettingsFile.startsWith("http")) {
-                    globalSettingsFile = downloadSettingsFromUrl(this, globalSettingsFile, 'global-settings.xml')
-                }
-                sh "cp ${globalSettingsFile} \$M2_HOME/conf/settings.xml"
-            }
+        String configFiles = piperGoUtils.prepareConfigurations([PIPER_DEFAULTS].plus(script.commonPipelineEnvironment.getCustomDefaults()), ADDITIONAL_CONFIGS_FOLDER)
+        writeFile(file: "${METADATA_FOLDER}/${METADATA_FILE}", text: libraryResource(METADATA_FILE))
```

Comment:
This is different from how checkmarxExecuteScan.groovy does it, where it ends up in <workspace root>/metadata/checkmarx.yml. It looks like putting it inside .pipeline/ is more clean, as that is already a reserved location in the project structure. Can someone advice, I would then change the nexusUpload.groovy which I've modeled after the checkmarx example.

Reply:
I can also put into `metadata`. IMO having it inside `pipeline` seems to be a better location since that path is already "reserved" by the our shared lib.

Reply:
Makes sense. I don't know whether Oliver's comment referred to the same thing. If nothing else, we can make it consistent across Groovy-wrapper-steps in a separate PR.